packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -0,0 +1,357 @@
+  ]);
+});
+
+test('solver distribute from one (A -> B 60, A -> C 40)', async t => {
+  const current = balances({ [A]: 100n, [B]: 0n, [C]: 0n });
+  const target = { [A]: ZERO, [B]: token(60n), [C]: token(40n) };
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target,
+    brand: TOK_BRAND,
+    links: LINKS,
+    mode: 'cheapest',
+  });
+  t.deepEqual(steps, [
+    { src: A, dest: '@Arbitrum', amount: token(100n) },
+    { src: '@Arbitrum', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(60n) },
+    { src: '@noble', dest: '@Ethereum', amount: token(40n) },
+    { src: '@Avalanche', dest: B, amount: token(60n) },
+    { src: '@Ethereum', dest: C, amount: token(40n) },
+  ]);
+});
+
+test('solver collect to one (B 30 + C 70 -> A)', async t => {
+  const current = balances({ [A]: 0n, [B]: 30n, [C]: 70n });
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target: { [A]: token(100n), [B]: ZERO, [C]: ZERO },
+    brand: TOK_BRAND,
+    links: LINKS,
+    mode: 'cheapest',
+  });
+  t.deepEqual(steps, [
+    { src: B, dest: '@Avalanche', amount: token(30n) },
+    { src: '@Avalanche', dest: '@noble', amount: token(30n) },
+    { src: C, dest: '@Ethereum', amount: token(70n) },
+    { src: '@Ethereum', dest: '@noble', amount: token(70n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(100n) },
+    { src: '@Arbitrum', dest: A, amount: token(100n) },
+  ]);
+});
+
+test('solver deposit redistribution (Deposit 100 -> A 70, B 30)', async t => {
+  const current = balances({ '<Deposit>': 100n, [A]: 0n, [B]: 0n });
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target: { '<Deposit>': ZERO, [A]: token(70n), [B]: token(30n) },
+    brand: TOK_BRAND,
+    links: LINKS,
+    mode: 'cheapest',
+  });
+  t.deepEqual(steps, [
+    { src: '<Deposit>', dest: '@agoric', amount: token(100n) },
+    { src: '@agoric', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(70n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(30n) },
+    { src: '@Arbitrum', dest: A, amount: token(70n) },
+    { src: '@Avalanche', dest: B, amount: token(30n) },
+  ]);
+});
+
+test('solver move pools to cash (A 50 + B 30 -> Cash)', async t => {
+  const current = balances({ [A]: 50n, [B]: 30n, '<Cash>': 0n });
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target: { [A]: ZERO, [B]: ZERO, '<Cash>': token(80n) },
+    brand: TOK_BRAND,
+    links: LINKS,
+    mode: 'cheapest',
+  });
+  t.deepEqual(steps, [
+    { src: A, dest: '@Arbitrum', amount: token(50n) },
+    { src: '@Arbitrum', dest: '@noble', amount: token(50n) },
+    { src: B, dest: '@Avalanche', amount: token(30n) },
+    { src: '@Avalanche', dest: '@noble', amount: token(30n) },
+    { src: '@noble', dest: '@agoric', amount: token(80n) },
```

Comment:
I think resolver should be able to handle the resolving two pending txns that are merged into one but we need to double check